lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -124,16 +140,24 @@ def exception_class_name(exception=@exception)
         end
 
         def failure_lines
-          @failure_lines ||=
-            begin
-              lines = []
-              lines << failure_slash_error_line unless (description == failure_slash_error_line)
-              lines << "#{exception_class_name}:" unless exception_class_name =~ /RSpec/
-              encoded_string(exception.message.to_s).split("\n").each do |line|
-                lines << "  #{line}"
-              end
-              lines
-            end
+          @failure_lines ||= failure_slash_error_lines + exception_lines
+        end
+
+        def failure_slash_error_lines
+          read_failed_lines.tap do |lines|
+            least_indentation = lines.map { |line| line.match(/^[ \t]*/)[0] }.min
```

Comment:
I think I'd prefer this as:

``` ruby
least_indentation = lines.map { |line| line[/^[ \t]*/] }.min
```

Besides being shorter, I suspect that it's faster (though, I haven't benchmarked it).  There's no need for it to instantiate the `MatchData` object and there's not an extra `[0]` message send.